### ISSUE TITLE
Pre-shutdown

### ISSUE THREAD

**Issue by brussee** [Tribler/tribler#2665]
Upon shutting down allow for graceful completion of important tasks with a 'pre-shutdown'.
While doing this show the user the shutdown (progress) status and optionally a "Close now" button to abort everything.
This approach solves the current buggy shutdown procedure where deferred functions are canceled twice because it is undetermined what the shutdown order is and what must happen before actually shutting down.
Then, when the real shutdown is triggered twisted can shutdown etc. Kind of like a pre-start.

**Comment by synctext** [comment]
Earliest reporting of shutdown problems that still haunt us this month, #4404 .
